pcap4j-core/src/main/java/org/pcap4j/packet/IcmpV6MobilePrefixAdvertisementPacket.java

```diff
@@ -0,0 +1,333 @@
+                            IpV6NeighborDiscoveryOptionType.class).newInstance(rawData, currentOffsetInHeader + offset,
+                            length - currentOffsetInHeader, type);
+                } catch (Exception e) {
+                    break;
+                }
+                options.add(newOne);
+                currentOffsetInHeader += newOne.length();
+            }
+        }
+
+        private IcmpV6MobilePrefixAdvertisementHeader(Builder builder) {
+            if ((builder.reserved & 0xC0000000) != 0) {
+                throw new IllegalArgumentException(
+                        "Invalid reserved: " + builder.reserved);
+            }
+            this.identifier = builder.identifier;
+            this.managedAddressConfigurationFlag = builder.managedAddressConfigurationFlag;
+            this.otherStatefulConfigurationFlag = builder.otherStatefulConfigurationFlag;
+            this.reserved = builder.reserved;
+            this.options = new ArrayList<IpV6NeighborDiscoveryOption>(builder.options);
+        }
+
+        public short getIdentifier() {
+            return identifier;
+        }
+
+        public boolean isManagedAddressConfigurationFlag() {
+            return managedAddressConfigurationFlag;
+        }
+
+        public boolean isOtherStatefulConfigurationFlag() {
+            return otherStatefulConfigurationFlag;
+        }
+
+        public short getReserved() {
+            return reserved;
+        }
+
+        public List<IpV6NeighborDiscoveryOption> getOptions() {
+            return new ArrayList<IpV6NeighborDiscoveryOption>(options);
+        }
+
+        @Override
+        protected List<byte[]> getRawFields() {
+            List<byte[]> rawFields = new ArrayList<byte[]>();
+
+            rawFields.add(ByteArrays.toByteArray(identifier));
+            short tmp = (short) (0x3FFF & reserved);
+            if (managedAddressConfigurationFlag) {
+                tmp |= 0x8000;
+            }
+            if (otherStatefulConfigurationFlag) {
+                tmp |= 0x4000;
+            }
+            rawFields.add(ByteArrays.toByteArray(tmp));
+            for (IpV6NeighborDiscoveryOption o : options) {
+                rawFields.add(o.getRawData());
+            }
+            return rawFields;
+        }
+
+        @Override
+        protected int calcLength() {
+            int len = 0;
+            for (IpV6NeighborDiscoveryOption o : options) {
+                len += o.length();
+            }
+            return len + OPTIONS_OFFSET;
+        }
+
+        @Override
+        protected String buildString() {
+            StringBuilder sb = new StringBuilder();
+            String ls = System.getProperty("line.separator");
+
+            sb.append("[ICMPv6 Mobile Prefix Advertisement Header (")
+                    .append(length())
+                    .append(" bytes)]")
+                    .append(ls);
+            sb.append("  Identifier: ")
+                    .append(identifier)
```

Comment:
Please add a return value of getIdentifierAsInt() instead of identifier for consistency with IcmpIdentifiablePacket.

Reply:
fixed.